src/Common/AsyncLoader.cpp

```diff
@@ -691,54 +701,51 @@ void AsyncLoader::wait(std::unique_lock<std::mutex> & job_lock, const LoadJobPtr
         if (worker_pool == job->pool_id)
         {
             job_lock.unlock(); // Avoid reverse locking order
-            workerIsSuspendedByWait(worker_pool, job);
+            std::unique_lock lock{mutex};
             job_lock.lock();
+
+            // Rechecks are required because we have reacquired mutexes
```

Comment:
`std::unique_lock job_lock{job->mutex};` not needed ?

Reply:
It also prevents unnecessary unlock/lock that was before the change due to code was inside the function